src/content/learn/referencing-values-with-refs.md

```diff
@@ -248,66 +249,69 @@ function useRef(initialValue) {
+- **ref को एक एस्केप हैच के रूप में इस्तेमाल करें।** जब आप एक्सटर्नल सिस्टम या ब्राउज़र APIs के साथ काम करते हैं तब ref बहुत काम आता हैं। यदि आपके एप्लिकेशन लॉजिक और डेटा फ्लो ref पर बहुत निर्भर करते हैं, तो आपको अपने एप्रोच को बदलने की ज़रूरत है।
 
-Limitations of React state don't apply to refs. For example, state acts like a [snapshot for every render](/learn/state-as-a-snapshot) and [doesn't update synchronously.](/learn/queueing-a-series-of-state-updates) But when you mutate the current value of a ref, it changes immediately:
+- **रेंडरिंग के दौरान `ref.current` को न पढ़ें और न ही लिखें।** अगर कुछ जानकारी की रेंडरिंग के दौरान ज़रूरत पड़ती है तो, [state](/learn/state-a-components-memory) का इस्तेमाल करें। क्योंकि React को पता नहीं होता कब `ref.current` बदलता है, यहाँ तक कि रेंडरिंग के दौरान इसे पढ़ने से आपके कौम्पोनॅन्ट के बिहेवियर को प्रेडिक्ट करना मुश्किल हो जाता है। (इसका एक ही एक्सेप्शन है जो आप `if (!ref.current) ref.current = new Thing()` ऐसे कोड का इस्तेमाल करके पहले रेंडर के दौरान रेफरेंस को सेट कर सकते हैं।)
+
+React state की सीमाएँ ref के लिए लागू नहीं होती हैं। उदाहरण के लिए, state [हर रेंडर के लिए एक स्नैपशॉट](/learn/state-as-a-snapshot) की तरह काम करता है और [सिंक्रोनोसली से अपडेट नहीं होता है।](/learn/queueing-a-series-of-state-updates) लेकिन जब आप ref के करंट वैल्यू को म्यूटेट करते हैं, तो वाह तुरंत बदल जाता है।
 
 ```js
 ref.current = 5;
 console.log(ref.current); // 5
-```
+``` 
 
-This is because **the ref itself is a regular JavaScript object,** and so it behaves like one.
+यह इसलिए होता है क्योंकि **ref खुद एक साधारण जावास्क्रिप्ट ऑब्जेक्ट है** और इसलिए यह उसके जैसे काम करता है।
 
-You also don't need to worry about [avoiding mutation](/learn/updating-objects-in-state) when you work with a ref. As long as the object you're mutating isn't used for rendering, React doesn't care what you do with the ref or its contents.
+जब आप ref के साथ काम करते हैं तो [म्यूटेशन से बचने](/learn/updating-objects-in-state) की चिंता करने की ज़रुरत नहीं है। जब तक आप म्युटेट कर रहे ऑब्जेक्ट को रेंडरिंग के लिए नहीं इस्तेमाल कर रहे हैं, React को कोई फर्क नहीं पड़ता कि आप ref या उसकी कंटेंट्स के साथ क्या कर रहे हैं।
 
-## Refs and the DOM {/*refs-and-the-dom*/}
+## ref और DOM {/*refs-and-the-dom*/}
 
 You can point a ref to any value. However, the most common use case for a ref is to access a DOM element. For example, this is handy if you want to focus an input programmatically. When you pass a ref to a `ref` attribute in JSX, like `<div ref={myRef}>`, React will put the corresponding DOM element into `myRef.current`. You can read more about this in [Manipulating the DOM with Refs.](/learn/manipulating-the-dom-with-refs)
```

Comment:
Lets remove this line. I think english version is also present with hindi version below

Reply:
yaa, I removed it now! I think I have forgotten to remove it last time, Sorry for that!